packages/api/endpoints/providers.js

```diff
@@ -55,25 +59,33 @@ async function get(req, res) {
  * @returns {Promise<Object>} the promise of express response object
  */
 async function post(req, res) {
-  const data = req.body;
-  const id = data.id;
-
-  const providerModel = new models.Provider();
-
   try {
-    // make sure the record doesn't exist
-    await providerModel.get({ id });
-    return res.boom.badReqest(`A record already exists for ${id}`);
-  } catch (e) {
-    if (e instanceof RecordDoesNotExist) {
-      const record = await providerModel.create(data);
-
-      if (inTestMode()) {
-        await addToLocalES(record, indexProvider);
+    const data = req.body;
+    const id = data.id;
+
+    const providerModel = new Provider();
+
+    try {
+      // make sure the record doesn't exist
+      await providerModel.get({ id });
+      return res.boom.conflict(`A record already exists for ${id}`);
+    } catch (e) {
+      if (e instanceof RecordDoesNotExist) {
+        const record = await providerModel.create(data);
+
+        if (inTestMode()) {
+          await addToLocalES(record, indexProvider);
+        }
+        return res.send({ record, message: 'Record saved' });
       }
-      return res.send({ record, message: 'Record saved' });
+      throw e;
+    }
+  } catch (err) {
+    if (isBadRequestError(err)) {
```

Comment:
I'm debating whether creating a helper to catch the common "bad request" errors is a good or a bad thing